staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/interface.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"time"
+
+	"github.com/google/cel-go/common/types/ref"
+
+	v1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/admission/plugin/webhook/generic"
+)
+
+var _ ExpressionAccessor = &MatchCondition{}
+
+type ExpressionAccessor interface {
+	GetExpression() string
+}
+
+// EvaluationResult contains the minimal required fields and metadata of a cel evaluation
+type EvaluationResult struct {
+	EvalResult ref.Val
+	Expression string
+	Elapsed    time.Duration
+	Error      error
+}
+
+// MatchCondition contains the inputs needed to compile, evaluate and match a cel expression
+type MatchCondition struct {
+	Expression string
+}
+
+func (v *MatchCondition) GetExpression() string {
+	return v.Expression
+}
+
+// FilterCompiler contains a function to assist with converting types and values to/from CEL-typed values.
+type FilterCompiler interface {
+	// Compile is used for the cel expression compilation
+	Compile(expressions []ExpressionAccessor, hasParam bool) Filter
+}
+
+// Filter contains a function to evaluate compiled CEL-typed values
+// It expects the inbound object to already have been converted to the version expected
+// by the underlying CEL code (which is indicated by the match criteria of a policy definition).
+// versionedParams may be nil.
+type Filter interface {
```

Comment:
I agree. I don't think validation applies very well in the case of match conditions. I like the name `Evaluator` for this, but I don't want to block moving this PR forward on naming.

Reply:
I agree that we should not use the term "validation" since the new use cases are not doing validation, per se. `evaluation` is maybe slightly better, but it's kinda vague. The two "types" of CEL expressions that have emerged so far are ones that evaluate to bool and ones that evaluate to string.  `evaluation` doesn't disambiguate between those two types of CEL expressions. So I'm more in favor of a term like "predicate" or "filter" or something that hints at the "boolean result" property of the CEL expression.

Reply:
Should I address this rename or handle renaming later per @tallclair 's suggestion?

Reply:
Let's merge this as-is and we can sort out the naming in a future PR.